### PR TITLE
fix(discover) Fix query parsing and expanded view URL generation

### PR DESCRIPTION
**Pull request by markstory**
Our query string parser was incorrectly handling quoted strings in tag values. This resulted in broken conditions being generated when searches included a quoted string in the first token.

I've also fixed the query generation for tag links in the details view. These conditions would duplicate conditions of a field was in both the selected results and condition string. Now the inferred value overwrites the previous condition.

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/tokenizeSearch.tsx [hunk 1]
```diff
@@ -28,24 +28,30 @@ export function tokenizeSearch(query: string) {
...
-    const tokenChars = Array.from(token);
     let tokenState: 'query' | 'tags' = 'query';
 
-    tokenChars.forEach((char, idx) => {
-      const nextChar = tokenChars.length > idx ? tokenChars[idx + 1] : null;
-
-      if (idx === 0 && [':', ' '].includes(char)) {
-        return;
-      }
+    // Traverse the token and determine if it is a tag
```
Comment (dashed) [inline src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: @markstory i thought this is a 1:1 translation from the python implementation. Does this mean the python implementation is buggy as well?

Comment (markstory) [reply src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: I've updated this code to be laid out the same as the python version.

Comment (dashed) [reply src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: @EvanPurkhiser

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: oops :)

**Review by evanpurkhiser — APPROVED** [review]
Solid fixes here.

File: src/sentry/static/sentry/app/utils/tokenizeSearch.tsx [hunk 2]
```diff
@@ -28,24 +30,28 @@ export function tokenizeSearch(query: string) {
...
-    tokenChars.forEach((char, idx) => {
-      const nextChar = tokenChars.length > idx ? tokenChars[idx + 1] : null;
+    // Traverse the token and determine if it is a tag
+    // condition or bare words.
+    for (let i = 0, len = token.length; i < len; i++) {
+      const char = token[i];
 
-      if (idx === 0 && [':', ' '].includes(char)) {
-        return;
+      if (i === 0 && char === '"') {
```
Comment (dashed) [inline src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]:
```suggestion
      if (i === 0 && (char === '"' || char === ':')) {
```

We also want to bail out on tokens that begin with a colon. These tokens are still queries.

Test case:
```js
      {
        name: 'tokens that begin with a colon are still queries',
        string: 'country:canada :unresolved',
        object: {query: [':unresolved'], country: ['canada']},
      },
```

**Review by dashed — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/utils/tokenizeSearch.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: Fixed now in both the JS and python implementation.

Comment (dashed) [reply src/sentry/static/sentry/app/utils/tokenizeSearch.tsx]: Oh interesting. I thought the Python covered this case. I had searched with a query that began with a colon on SaaS .

**Review by dashed — APPROVED** [review]
Looks good to me. 👍  Just need to fix a TypeScript error.

**Pull request merged**